### DISCUSSION TITLE
Make Bonus Islands unlockable through Tux Dolls

### DISCUSSION THREAD

**Discussion by Rusty-Box** [SuperTux/supertux#2595] [Ideas]
With the removal of lives, Tux Dolls and coins were kinda left in as is despite not providing much gameplay value. Coins at the very least still provide some sort of purpose by being a requirement to collect all in a level for it to count as 100% completed.

Tux Dolls however were literally useless since then. The idea I am proposing here may make them a bit more special again.

Lock the Bonus Islands behind Tux Dolls
-----------------------------------------

The idea I am proposing is to make Tux Dolls required to unlock the Bonus Islands (not including add-ons like Revenge in Redmond or the Seasonal Specials). The way it would work is that you unlock a Bonus Island after collecting all Tux Dolls in a single world.

For example, collecting all Tux Dolls in **Icy Island** will unlock **Bonus Island I**. Collecting all Tux Dolls in **Rooted Forest** will unlock **Bonus Island II**, and so on...

---

With v0.7 making it a level requirement to feature atleast one Tux Doll in every level, this would encourage players to properly look out for secrets (unless they are already going for 100% in the first place). Also, to make them less annoying to get, instead of coming from Bonus Blocks they should float freely in the air, similar to a Key.

When going into the official add-on selecting one can see the Bonus Islands but great out. Trying to access them before unlocking would pop up a message saying something along the way of:  "You must collect all Tux Dolls in World X to gain access!" - players should know the Bonus Islands exist and that they are unlockable!

**Comment by GamerOn117** [comment]
Interesting idea
If you do this, I recommend you:

- Add a stat to every level to show players how many Tux dolls are still uncollected
- Give the normal coin reward when collected the first time, then give a nerfed coin reward every time after (so as to not break shops)
- Allow a way to the bonus islands via the main world so there's no exiting the main campaign to play them. (Wait, there are five bonus islands and two main worlds. What happens to the other three?)
- Give a reward for the campaign if you beat the bonus islands, maybe similar to the switch palaces in Mario World
If I come up with more, I'll edit my comment.

**Comment by provigz** [comment]
This is a nice idea for story mode. However, how would other worlds implement this?

I was made aware that there has been the idea of allowing customization of the behaviour of Tux Doll for other worlds, via scripting. However, this seems to make it, as a base game mechanic, quite inconsistent throughout the game, and it's all up to the designer to make a purpose for it, which would usually not be expected for base game mechanics.

Another topic I have been thinking about over time, which would be up for discussion, is giving purpose to coins. Since the checkpoint coin limit mechanic was removed, coins do not serve a realistic purpose in any worlds in the game. A possible solution, which has been suggested for that, is the inclusion of shops, but that seems to only solve the issue for story mode, hence I don't believe it is a general fix to the issue.

Finding a global purpose to coins in the game should technically solve this issue as well, since Tux Dolls giving 100 coins would be pretty welcome, in my opinion.

**Comment by Rusty-Box** [comment]
Tux Dolls could get their own slot in the level statistics as a requirement for 100% a level. Think of them like star coins from the NSMB series. A simple collectable to be used in level, both official and custom.

If a level/world creator desires to give players a special reward for collecting every single Tux within their add-on, there should be methods in place to allow that. Let them unlock their own bonus worlds, level pack etc. If nothing is added, they will just remain a simple collectible.
